packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -886,3 +886,379 @@ RAW_TOOLS_DIR: ${Env:RAW_TOOLS_DIR}
+        [string]$type,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$path
+    )
+
+    try {
+        if ($type -eq "file") {
+            if (Test-Path -Path $path -PathType Leaf) {
+                Remove-Item -Path $path -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        }
+        elseif ($type -eq "dir") {
+            if (Test-Path -Path $path -PathType Container) {
+                Remove-Item -Path $path -Recurse -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        }
+    }
+    catch {
+        VM-Write-Log "ERROR" "An error occurred while removing the $type $path. Error: $_"
+    }
+}
+
+function VM-Execute-Custom-Command{
+# Function for removing items in need of custom code.
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string[]]$cmds
+    )
+
+    try {
+        Write-Output "[+] Executing commands for '$name':"
+        foreach ($cmd in $cmds) {
+            Write-Output "`t[+] Executing command: $cmd"
+            start-process powershell -ArgumentList "-WindowStyle","Hidden","-Command",$cmd -Wait
+            Write-Host "`t[+] Process completed. Moving to next."
+        }
+        Write-Output "[+] All commands for '$name' have been executed successfully."
+    }
+    catch {
+        VM-Write-Log "ERROR" "An error occurred while executing commands for '$name'. Error: $_"
+    }
+}
+
+function VM-Configure-Prompts {
+    # $Env:MandiantVM must be set in the install script
+    $psprompt = @"
+        function prompt {
+            Write-Host (`$Env:MandiantVM + " " + `$(Get-Date)) -ForegroundColor Green
+            Write-Host ("PS " + `$(Get-Location) + " >") -NoNewLine -ForegroundColor White
+            return " "
+        }
+"@
+
+    # Ensure profile file exists and append new content to it, not overwriting old content
+    if (!(Test-Path $profile)) {
+        New-Item -ItemType File -Path $profile -Force | Out-Null
+    }
+    Add-Content -Path $profile -Value $psprompt
+
+    # Add timestamp to cmd prompt
+    ## Configure the command
+    $mandiantVM = $Env:MandiantVM -replace ' ', '' # setx command cannot have spaces
+    $command = "cmd /c 'setx PROMPT $mandiantVM`$S`$d`$s`$t`$_`$p$+`$g'"
+    ## Convert to base64
+    $base64 = [Convert]::ToBase64String([System.Text.Encoding]::Unicode.GetBytes($command))
+    ## Run command
+    Invoke-Expression ([System.Text.Encoding]::Unicode.GetString([System.Convert]::FromBase64String($base64))) | Out-Null
+
+    Write-Host "[+] Timestamps added to cmd prompt and PowerShell" -ForegroundColor Green
```

Comment:
Let's swap out the generic "Write-Host" values with "VM-Write-Log" where possible, otherwise we won't log various messages

Reply:
There's a few sprinkled around this PR, only adding this one comment to account for all of the instances

Reply:
I'll take a look around, some of them I left out intentionally so I will make another pass